libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/MapboxMetricsReporter.kt

```diff
@@ -0,0 +1,101 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation.metrics
+
+import android.content.Context
+import com.google.gson.Gson
+import com.mapbox.android.telemetry.AppUserTurnstile
+import com.mapbox.android.telemetry.MapboxTelemetry
+import com.mapbox.navigation.metrics.DirectionsMetrics
+import com.mapbox.navigation.metrics.MetricsObserver
+import com.mapbox.navigation.metrics.NavigationMetrics
+import com.mapbox.services.android.navigation.v5.internal.navigation.InitialGpsEvent
+import com.mapbox.services.android.navigation.v5.internal.navigation.RouteRetrievalEvent
+import java.util.concurrent.atomic.AtomicReference
+
+internal object MapboxMetricsReporter : DirectionsMetrics, NavigationMetrics {
```

Comment:
should this class live in `libnavigation-metrics`?

Reply:
As I thought `libnavigation-metrics` is module that define base of general metrics stuff. And MapboxMetricsReporter - concrete realization of our mapbox metrics reporting, so it should belong to `libandroid-navigation` module. But yeah, I free to discuss

Reply:
`MapboxMetricsReporter` is our default implementation, that's right. But in order for it to be replaceable, it should live in a separate module. For example, `libnavigation-metrics` like @andrlee mentioned.

More so,  `DirectionsMetrics`, `NavigationMetrics` and `MetricsObserver` should live in the `libnavigation-base`, so that our default implementation and developer's possible replacement have a common place to extend from.